contracts/registry/Registry.sol

```diff
@@ -30,13 +41,39 @@ contract Registry is AdminRole {
     /// @dev Emit when a contributor has been removed:
     event ContributorRemoved(address adr);
 
+    /// @dev Emit when a contributor's pending balance is changed:
+    event PendingBalanceChanged(address indexed adr, uint256 pendingBalance);
+
+    /// @dev Emit when a contributor's pending balance is cleared:
+    event PendingBalanceCleared(
+        address indexed adr,
+        uint256 consumedPendingBalance
+    );
+
+    /// @dev Emit when minter contract address is set
+    event MinterContractSet(address indexed adr);
+
     //
     // CONSTRUCTOR:
     //
 
     /// @dev Construct the Registry,
     /// @param _admins (address[]) List of admins for the Registry contract.
-    constructor(address[] memory _admins) public AdminRole(_admins) {}
+    /// @param _cstkTokenAddress (address) CS token deployed contract address
+    constructor(address[] memory _admins, address _cstkTokenAddress)
+        public
+        AdminRole(_admins)
+    {
+        cstkToken = IERC20(_cstkTokenAddress);
+    }
+
+    modifier onlyMinter() {
+        require(
+            _msgSender() == minterContract,
+            "Caller is not Minter Contract"
+        );
+        _;
+    }
```

Comment:
Should we add pendingbalance to `registerContributor` and `registerContributors`?

Reply:
It's initial value is considered to be zero now, nevertheless if again it's safe I feel better if use can set it in registring.

Reply:
You mean to speed up the process Kristofer?

Reply:
Again, I find only use in the support scripts and test cases. 

https://github.com/commons-stack/cstk-token/blob/94a73886246500ddf155080b8f778fecf44d5c1a/scripts/addTrusted.ts#L17

https://github.com/commons-stack/cstk-token/blob/94a73886246500ddf155080b8f778fecf44d5c1a/test/unit/Registry.spec.ts#L47

Reply:
> You mean to speed up the process Kristofer?

Partly, but also to have a "full" implementation. If the registry consist of "account, maxTrust, pendingBalance", it sort of makes sense to include all three when registering contributors.